Makefile

```diff
@@ -302,34 +313,57 @@ virtualenv:
 	@echo
 	# Note: We pass --no-download flag to make sure version of pip which we install (9.0.1) is used
 	# instead of latest version being downloaded from PyPi
-	test -f $(VIRTUALENV_DIR)/bin/activate || virtualenv --python=$(PYTHON_VERSION) --no-site-packages $(VIRTUALENV_DIR) --no-download
-
-	# Setup PYTHONPATH in bash activate script...
-	# Delete existing entries (if any)
-	sed -i '/_OLD_PYTHONPATHp/d' $(VIRTUALENV_DIR)/bin/activate
-	sed -i '/PYTHONPATH=/d' $(VIRTUALENV_DIR)/bin/activate
-	sed -i '/export PYTHONPATH/d' $(VIRTUALENV_DIR)/bin/activate
-
-	echo '_OLD_PYTHONPATH=$$PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate
-	#echo 'PYTHONPATH=$$_OLD_PYTHONPATH:$(COMPONENT_PYTHONPATH)' >> $(VIRTUALENV_DIR)/bin/activate
-	echo 'PYTHONPATH=${ROOT_DIR}:$(COMPONENT_PYTHONPATH)' >> $(VIRTUALENV_DIR)/bin/activate
-	echo 'export PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate
-	touch $(VIRTUALENV_DIR)/bin/activate
-
-	# Setup PYTHONPATH in fish activate script...
-	#echo '' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo 'set -gx _OLD_PYTHONPATH $$PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo 'set -gx PYTHONPATH $$_OLD_PYTHONPATH $(COMPONENT_PYTHONPATH)' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo 'functions -c deactivate old_deactivate' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo 'function deactivate' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '  if test -n $$_OLD_PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '    set -gx PYTHONPATH $$_OLD_PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '    set -e _OLD_PYTHONPATH' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '  end' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '  old_deactivate' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo '  functions -e old_deactivate' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#echo 'end' >> $(VIRTUALENV_DIR)/bin/activate.fish
-	#touch $(VIRTUALENV_DIR)/bin/activate.fish
+	# For OS X, we have to bootstrap install pip because otherwise it runs into
+	# SSLv1 errors when connecting to https://pypi.python.org/simple/ to
+	# download anything
+	@if [[ "$(OS)" == "Darwin" ]]; then \
+	    if [[ ! -f $(VIRTUALENV_DIR)/bin/activate ]]; then \
+	        virtualenv --python=$(PYTHON_VERSION) --no-site-packages --no-setuptools --never-download $(VIRTUALENV_DIR); \
```

Comment:
I'd install pip first and then virtualenv because stock OS X doesn't come with neither. We can also assume people have pip and virtualenv installed in which case you should just check if they exist and if not ask user to install those.

Reply:
The situation here is a little more complicated. This code actually does assume that the user has some version of virtualenv installed, and the most common way to install virtualenv is with pip, so if users have virtualenv, they probably also have *a version of* pip.

The issue with pip on OS X is that you might run into [this issue](https://github.com/pypa/pip/issues/5236), where you need a recent version of pip to install or upgrade anything *including pip itself*. So if you have a version of pip that is out of date, you cannot use it to do anything.

The problem is that you need to install a recent version of pip into the virtualenv, but you cannot use pip to do it. The method I use to install/upgrade pip here is pretty bulletproof (although it its a [little scary from a security perspective](https://www.idontplaydarts.com/2016/04/detecting-curl-pipe-bash-server-side/)) and is an [official way to install pip](https://pip.pypa.io/en/stable/installing/#installing-with-get-pip-py).